cmd/template/nodepool/runner.go

```diff
@@ -123,3 +133,41 @@ func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) err
 
 	return nil
 }
+
+func getWorkloadCluster(ctx context.Context, client *dataClient.Client, namespace, name, provider string) (*cluster.Cluster, error) {
+	var err error
+
+	var clusterService cluster.Interface
+	{
+		serviceConfig := cluster.Config{
+			Client: client,
+		}
+		clusterService, err = cluster.New(serviceConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	o := cluster.GetOptions{
+		Namespace: namespace,
+		Name:      name,
+		Provider:  provider,
+	}
+
+	c, err := clusterService.Get(ctx, o)
+	if cluster.IsNotFound(err) || cluster.IsNoResources(err) {
+		return nil, microerror.Maskf(clusterNotFoundError, "The workload cluster %s could not be found.", name)
+	} else if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	switch wc := c.(type) {
+	case *cluster.Cluster:
+		return wc, nil
+	case *cluster.Collection:
+		targetWC := wc.Items[0]
+		return &targetWC, nil
```

Comment:
Not sure if `&wc.Items[0]` would be safe here, as I believe `wc` would go out of scope and be garbage collected. Or would it be borrowed?

Reply:
maybe adding a test for this function would make sense, wdyt?

Reply:
Added one now, please take a look